packages/code/src/Code.tsx

```diff
@@ -140,8 +224,34 @@ function Code({
   chromeTitle = '',
   ...rest
 }: CodeProps) {
+  const [copied, setCopied] = useState(false);
+  const [canScrollLeft, setCanScrollLeft] = useState(false);
+  const [canScrollRight, setCanScrollRight] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const scrollableMultiLine = useRef<HTMLPreElement>(null);
+  const scrollableSingleLine = useRef<HTMLDivElement>(null);
+  useEffect(() => {
```

Comment:
The reason I have a useEffect here is to force a rerender after the initial render. We cannot determine if the Code content overflows its container until after the initial render, so it is impossible to know until then if a shadow should be shown on the right side of the container. Brooke mentioned that this was kind of a hack-y solution, but for now it gets the job done. I would love to know if y'all have better ideas for how to accomplish showing the shadow initially.

Reply:
The reason makes sense. You need to read the DOM before you can make decisions. There's a different hook just for this need, [useLayoutEffect](https://reactjs.org/docs/hooks-reference.html#uselayouteffect).